libdnf/dnf-sack.c

```diff
@@ -295,8 +295,23 @@ dnf_sack_recompute_considered(DnfSack *sack)
         map_subtract(pool->considered, priv->repo_excludes);
     if (priv->pkg_excludes)
         map_subtract(pool->considered, priv->pkg_excludes);
-    if (priv->pkg_includes)
-        map_and(pool->considered, priv->pkg_includes);
+    if (priv->pkg_includes) {
+        Map pkg_includes_tmp;
+        map_init_clone(&pkg_includes_tmp, priv->pkg_includes);
+        Id repoid;
+        Repo *repo;
+        FOR_REPOS(repoid, repo) {
+            HyRepo hyrepo = repo->appdata;
+            if (/*!repo->disabled &&*/ !hy_repo_get_includes_enabled(hyrepo)) {
```

Comment:
We talked about searching for packages in disabled repos some time ago. I'm afraid that enabling of this optimization can complicate this feature. On the other hand do we support searching in disabled repos?